Update to Sphinx 7

drasmuss opened the pull request.

drasmuss committed:
Update to Sphinx 7

drasmuss committed:
Remove backoff extension

This behaviour is now built into Sphinx.

tbekolay reviewed (commented):
All of the extensions seem to be working, which is good, but some of our custom CSS is not working in this branch (but it works in main). Hopefully it's just a matter of updating some selectors, I'll take a look.

tbekolay committed:
Fix CSS and move CSS file here

Previously the CSS file was served through nengo.ai.
Moving it here simplifies future development, and ensures
that future changes to CSS don't affect older doc builds.

tbekolay committed:
Remove some styles specific to nengo.ai

tbekolay commented:
Alrighty I pushed the CSS fixes. A few things had changed in the HTML rendered from CSS, and I moved the CSS file here so we didn't have to keep track of what is there for compatibility with previous versions of Sphinx.

I spent a fair bit of time trying to minimize the size of the CSS file, but ultimately wasn't very successful, as we were apparently shipping a customized version of bootstrap, so we couldn't work off the CDN delivered version, and removing bits of unused CSS might trip us up later when we look at the Bootstrap docs to figure out what we can use. It would have been a small optimization in any case, `nengo.css` is 206kb so in the best case we maybe save 100kb.
